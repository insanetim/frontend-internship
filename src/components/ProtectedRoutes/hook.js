import { useEffect } from 'react'
import * as Cookies from 'js-cookie'
import { useDispatch } from 'react-redux'

import { requestAccount, setSession } from 'src/store/actions'

export const useContainer = () => {
  const dispatch = useDispatch()
  const sessionId = Cookies.get('session_id')

  useEffect(() => {
    dispatch(setSession(sessionId || ''))
    if (sessionId) {
      dispatch(requestAccount())
    }
  }, [sessionId])

  return { sessionId }
}
