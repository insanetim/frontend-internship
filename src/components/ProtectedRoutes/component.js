import React from 'react'
import { Outlet, Navigate, useLocation } from 'react-router-dom'

import { useContainer } from './hook'

const ProtectedRoutes = () => {
  const location = useLocation()
  const { sessionId } = useContainer()

  return sessionId ? (
    <Outlet />
  ) : (
    <Navigate
      to='/login'
      replace
      state={{ from: location }}
    />
  )
}

export default ProtectedRoutes
