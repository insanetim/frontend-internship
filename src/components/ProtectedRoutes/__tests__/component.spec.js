/* eslint-disable react/jsx-no-useless-fragment */
import { shallow } from 'enzyme'

import ProtectedRoutes from '../component'

jest.mock('react-router-dom', () => ({
  Outlet: () => <></>,
  Navigate: () => <></>,
  useLocation: jest.fn(() => ({}))
}))

const mockedHookData = {
  sessionId: 'session_id'
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

it('matches snapshot', () => {
  const component = shallow(<ProtectedRoutes />)

  expect(component).toMatchSnapshot()
})

it('matches snapshot without session_id', () => {
  mockedHookData.sessionId = null
  const component = shallow(<ProtectedRoutes />)

  expect(component).toMatchSnapshot()
})
