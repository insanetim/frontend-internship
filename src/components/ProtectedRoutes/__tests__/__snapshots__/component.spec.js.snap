// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<Outlet />`;

exports[`matches snapshot without session_id 1`] = `
<Navigate
  replace={true}
  state={
    {
      "from": {},
    }
  }
  to="/login"
/>
`;
